### PR TITLE
feat: add external cloud-provider-aws

### PR DESCRIPTION
**Pull request by AlekSi**
Closes talos-systems/talos#3312.

### CODE REVIEW

File: aws/README.md [hunk 1]
```diff
@@ -1,9 +1,22 @@
...
+- A cluster that uses autoscaling groups for the worker set. Autoscaling groups are called "MachinePools" in Cluster API lingo.
+
+We also provide different variants for Talos 0.10 and up that added better support for
+[external cloud providers](https://kubernetes.io/docs/tasks/administer-cluster/running-cloud-controller/),
+including [cloud-provider-aws](https://github.com/kubernetes/cloud-provider-aws).
+
+| Directory                                                       | Talos | cluster-api-provider-aws (CAPA) |
+| --------------------------------------------------------------- | ----- | ------------------------------- |
+| [standard](./standard/)                                         | 0.9+  | 0.5.4                           |
+| [standard-external](./standard-external/)                       | 0.9+  | 0.5.4                           |
```
Comment (AlekSi) [inline aws/README.md:15]: 0.5.4 is the version we actually test with: https://github.com/talos-systems/talos/blob/33035901ff7875bdf9eb99fb86b377318f60d74b/hack/test/e2e-capi.sh#L9

Should we mention it? Should we update it to 0.5.5? 0.6.4?

File: aws/README.md [hunk 2]
```diff
@@ -1,9 +1,22 @@
...
+| [autoscaling-workers](./autoscaling-workers/)                   | 0.10+ | 0.6.1+                          |
+| [autoscaling-workers-external](./autoscaling-workers-external/) | 0.10+ | 0.6.1+                          |
```
Comment (AlekSi) [inline aws/README.md:17]: It looks like 0.6.1 added support for autoscaling groups: https://github.com/kubernetes-sigs/cluster-api-provider-aws/releases

Should we mention that version? Or that should 0.6.4?

Should we actually test Talos with 0.6 on CI?

File: aws/README.md [hunk 3]
```diff
@@ -1,9 +1,22 @@
...
+
+- cloud-provider-aws: v1.20.0-alpha.0
```
Comment (AlekSi) [inline aws/README.md:19]: Should we mention it and k8s versions?

Should we test Talos with different versions on CI?

File: aws/README.md (hunk 1 above)
Comment (smira) [reply aws/README.md:15]: I would say 0.5.x might be easier, assuming that minor versions should be fine, @rsmitty?

File: aws/README.md [hunk 4]
```diff
@@ -1,9 +1,22 @@
...
+
+- A "standard" cluster with HA control plane and a machine deployment for workers.
+- A cluster that uses autoscaling groups for the worker set. Autoscaling groups are called "MachinePools" in Cluster API lingo.
+
+We also provide different variants for Talos 0.10 and up that added better support for
+[external cloud providers](https://kubernetes.io/docs/tasks/administer-cluster/running-cloud-controller/),
+including [cloud-provider-aws](https://github.com/kubernetes/cloud-provider-aws).
+
+| Directory                                                       | Talos | cluster-api-provider-aws (CAPA) |
+| --------------------------------------------------------------- | ----- | ------------------------------- |
```
Comment (smira) [inline aws/README.md:13]: should we do `v0.1/stantard/...` and `v0.2/standard/..` here saying that v0.2 is only for Talos v0.10+ with `cloudProvider: external` config

**Comment by Unix4ever** [comment]
Commit message should probably be made with `-s`. We don't have conform here, so it didn't complain :)

File: aws/README.md (hunk 4 above)
Comment (AlekSi) [reply aws/README.md:13]: The only reason I did not do that is my desire to avoid even more non-aligned version numbers… Will think about it.

**Review by smira — COMMENTED** [review]
I believe the last thing we talked about was not to use subdirectories, instead we decided to do the following (@rsmitty  please correct me if I'm wrong):

1. release what we have right now as `v0.1.0` (in release-0.9 of Talos, update template reference to v0.1.0 instead of main)
2. commit changes to `main` just updating current templates, this should work for 0.10 of Talos

**Comment by rsmitty** [comment]
Sounds right to me.

**Review by smira — APPROVED** [review]

**Review by rsmitty — APPROVED** [review]

**Pull request merged**